cime_config/stream_cdeps.py

```diff
@@ -54,16 +61,55 @@ def __init__(self, infile, schema):
         if os.path.exists(infile):
             GenericXML.read(self, infile, schema)
 
-    def create_stream_xml(self, stream_names, case, streams_xml_file, data_list_file, available_neon_data=None):
+    def create_stream_xml(self, stream_names, case, streams_xml_file, data_list_file, user_mods_file,
+                          available_neon_data=None):
         """
         Create the stream xml file and append the required stream input data to the input data list file
         available_neon_data is an optional list of NEON tower data available for the given case, if provided
         this data will be used to populate the NEON streamdata list
         """
+
+        # determine if there are user mods
+        lines_input = []
+        expect (os.path.isfile(user_mods_file),
+                "No file {} found in case directory".format(user_mods_file))
+        with open(user_mods_file, "r", encoding='utf-8') as stream_mods_file:
+            lines_input = stream_mods_file.readlines()
+        stream_mod_dict = {}
+        for line in lines_input:
+            # read in a single line in user_nl_xxx_streams and parse it if it is not a comment
+            if ('!' not in line ):
```

Comment:
I think that this should check for comment at the beginning of the line and not just any !